core/market/tests/test_app_session_id.rs

```diff
@@ -0,0 +1,549 @@
+    let network = MarketsNetwork::new(None)
+        .await
+        .add_market_instance(REQ_NAME)
+        .await?
+        .add_market_instance(PROV_NAME)
+        .await?;
+
+    let req_market = network.get_market(REQ_NAME);
+    let req_id = network.get_default_id(REQ_NAME);
+    let prov_id = network.get_default_id(PROV_NAME);
+    let prov_market = network.get_market(PROV_NAME);
+
+    let negotiation = negotiate_agreement(
+        &network,
+        REQ_NAME,
+        PROV_NAME,
+        "negotiation",
+        "r-session",
+        "p-session",
+    )
+    .await
+    .unwrap();
+
+    let confirm_timestamp = negotiation.confirm_timestamp;
+    let p_events = prov_market
+        .query_agreement_events(
+            &Some("p-session".to_string()),
+            1.0,
+            Some(10),
+            confirm_timestamp,
+            &prov_id,
+        )
+        .await
+        .unwrap();
+
+    let r_events = req_market
+        .query_agreement_events(
+            &Some("r-session".to_string()),
+            0.5,
+            Some(10),
+            confirm_timestamp,
+            &req_id,
+        )
+        .await
+        .unwrap();
+
+    // Each side gets only his own event.
+    assert_eq!(p_events.len(), 1);
+    assert_eq!(r_events.len(), 1);
+    Ok(())
+}
+
+/// Test case, when Provider and Requestor is on the same node.
+#[cfg_attr(not(feature = "test-suite"), ignore)]
+#[actix_rt::test]
+#[serial_test::serial]
+async fn test_session_negotiation_on_the_same_node() -> Result<()> {
+    let network = MarketsNetwork::new(None)
+        .await
+        .add_market_instance("Node")
+        .await?;
+
+    let req_market = network.get_market("Node");
+    let req_id = network.get_default_id("Node");
+    let prov_id = network.get_default_id("Node");
+    let prov_market = network.get_market("Node");
+
+    let negotiation = negotiate_agreement(
+        &network,
+        "Node",
+        "Node",
+        "negotiation",
+        "r-session",
+        "p-session",
+    )
+    .await
+    .unwrap();
+
+    let confirm_timestamp = negotiation.confirm_timestamp;
+    let p_events = prov_market
+        .query_agreement_events(
+            &Some("p-session".to_string()),
+            1.0,
+            Some(10),
+            confirm_timestamp,
+            &prov_id,
+        )
+        .await
+        .unwrap();
+
+    let r_events = req_market
+        .query_agreement_events(
+            &Some("r-session".to_string()),
+            0.5,
+            Some(10),
+            confirm_timestamp,
+            &req_id,
+        )
+        .await
+        .unwrap();
+
+    // Each side gets only his own event.
+    assert_eq!(p_events.len(), 1);
+    assert_eq!(r_events.len(), 1);
+    Ok(())
+}
+
+/// We expect to get only events after specified timestamp.
+#[cfg_attr(not(feature = "test-suite"), ignore)]
+#[actix_rt::test]
+#[serial_test::serial]
+async fn test_session_timestamp_filtering() -> Result<()> {
```

Comment:
wow, thats huge test. I dare to propose even more: test mixed timestamp & session filtering. but will understand if you are fed up with this ;)

Reply:
Yes I am :P
Added to task: #781